src/platform/plugins/shared/controls/public/actions/delete_control_action.tsx

```diff
@@ -84,6 +84,7 @@ export class DeleteControlAction implements Action<EmbeddableApiContext> {
   public async execute({ embeddable }: EmbeddableApiContext) {
     if (!compatibilityCheck(embeddable)) throw new IncompatibleActionError();
 
+    // esql_control_removed
```

Comment:
This is generic per control. You need the specific event for the esql type control